tests/subsys/net/lib/nrf_cloud/fota_common/src/main.c

```diff
@@ -0,0 +1,716 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <net/nrf_cloud.h>
+#include "fakes.h"
+
+/* This function runs before each test */
+static void run_before(void *fixture)
+{
+	ARG_UNUSED(fixture);
+}
+
+/* This function runs after each completed test */
+static void run_after(void *fixture)
+{
+	ARG_UNUSED(fixture);
+}
+
+ZTEST_SUITE(nrf_cloud_fota_common_test, NULL, NULL, run_before, run_after, NULL);
+
+ZTEST(nrf_cloud_fota_common_test, test_nrf_cloud_fota_is_type_modem)
+{
+	/* Test range beyond enum values */
+	for (enum nrf_cloud_fota_type fota_type = NRF_CLOUD_FOTA_TYPE__FIRST - 1;
+	     fota_type <= NRF_CLOUD_FOTA_TYPE__INVALID + 1;
+	     ++fota_type) {
+
+		if ((fota_type == NRF_CLOUD_FOTA_MODEM_FULL) ||
+		    (fota_type == NRF_CLOUD_FOTA_MODEM_DELTA)) {
+			zassert_true(nrf_cloud_fota_is_type_modem(fota_type),
+				     "Modem FOTA type not detected");
+		} else {
+			zassert_false(nrf_cloud_fota_is_type_modem(fota_type),
+				      "Non-modem FOTA type detected as modem type");
+		}
+	}
+}
+
+#define APP_FOTA_ASSERT_STR "APP FOTA enabled should correspond to the value of: " \
+			    "CONFIG_BOOTLOADER_MCUBOOT"
+
+#define BOOT_FOTA_ASSERT_STR "BOOT FOTA enabled should correspond to the value of: " \
+			     "CONFIG_BOOTLOADER_MCUBOOT + CONFIG_BUILD_S1_VARIANT + " \
+			     "CONFIG_SECURE_BOOT"
+
+#define MDM_DELTA_ASSERT_STR "DELTA MODEM FOTA enabled should correspond to the value of: " \
+			     "CONFIG_NRF_MODEM"
+
+#define FMFU_ASSERT_STR "FULL MODEM FOTA enabled should correspond to the value of: " \
+			"CONFIG_NRF_CLOUD_FOTA_FULL_MODEM_UPDATE + CONFIG_NRF_MODEM"
+
+ZTEST(nrf_cloud_fota_common_test, test_nrf_cloud_fota_is_type_enabled)
+{
+	Z_TEST_SKIP_IFNDEF(CONFIG_FOTA_DOWNLOAD);
+
+	bool ret;
+
+	ret = nrf_cloud_fota_is_type_enabled(NRF_CLOUD_FOTA_APPLICATION);
+	zassert_equal(IS_ENABLED(CONFIG_BOOTLOADER_MCUBOOT), ret, APP_FOTA_ASSERT_STR);
+
+	ret = nrf_cloud_fota_is_type_enabled(NRF_CLOUD_FOTA_BOOTLOADER);
+	zassert_equal(IS_ENABLED(CONFIG_BOOTLOADER_MCUBOOT) &&
+		      IS_ENABLED(CONFIG_BUILD_S1_VARIANT) &&
+		      IS_ENABLED(CONFIG_SECURE_BOOT),
+		      ret,
+		      BOOT_FOTA_ASSERT_STR);
+
+	ret = nrf_cloud_fota_is_type_enabled(NRF_CLOUD_FOTA_MODEM_DELTA);
+	zassert_equal(IS_ENABLED(CONFIG_NRF_MODEM), ret, MDM_DELTA_ASSERT_STR);
+
+	ret = nrf_cloud_fota_is_type_enabled(NRF_CLOUD_FOTA_MODEM_FULL);
+	zassert_equal(IS_ENABLED(CONFIG_NRF_CLOUD_FOTA_FULL_MODEM_UPDATE) &&
+		      IS_ENABLED(CONFIG_NRF_MODEM),
+		      ret,
+		      FMFU_ASSERT_STR);
+
+	ret = nrf_cloud_fota_is_type_enabled(NRF_CLOUD_FOTA_TYPE__INVALID);
+	zassert_false(ret, "return should be false when invalid FOTA type is provided");
```

Comment:
Each of these could arguably be a separate test case

Reply:
Some folks insist upon it, even when it makes the test cases significantly more complex! The idea is that each unit test should focus on a single minimal function. This theoretically makes the test easier to maintain, understand, and interpret in logs.

IMO splitting tests up like that is definitely desirable, but only worthwhile when it does not come at a significant cost to test complexity / readability / redundancy.

In this case I can't see any costs to doing it, so I lean toward splitting it

But this is another one of those situations where it's really not a big deal, so if you want to keep it as all one test, it isn't too huge of a bother for me